lib/topic_query.rb

```diff
@@ -680,7 +680,7 @@ def default_results(options = {})
         tags_query = tags_arg[0].is_a?(String) ? Tag.where_name(tags_arg) : Tag.where(id: tags_arg)
         tags = tags_query.select(:id, :target_tag_id).map { |t| t.target_tag_id || t.id }.uniq
 
-        if @options[:match_all_tags]
+        if ActiveModel::Type::Boolean.new.cast(@options[:match_all_tags])
```

Comment:
I kind of feel like type casting should be done at the controller level which deals with the params.

Reply:
I put the type casting at this lower level because I thought it would act as a sort of final sanity check against the values being passed down to the lib (rather than trying to identify all of the places the lib was being called from).